catalog/spatial/registry/registry-federation-admin-impl/src/main/java/org/codice/ddf/registry/federationadmin/impl/AdminHelper.java

```diff
@@ -70,11 +70,10 @@ public AdminHelper(
   }
 
   private BundleContext getBundleContext() {
-    Bundle bundle = FrameworkUtil.getBundle(this.getClass());
-    if (bundle != null) {
-      return bundle.getBundleContext();
-    }
-    return null;
+    return Optional.of(this.getClass())
+        .map(FrameworkUtil::getBundle)
+        .map(Bundle::getBundleContext)
+        .orElseThrow(() -> new IllegalStateException("Error getting bundle context"));
```

Comment:
❓ Same question as before from the outside perspective we are now throwing an IllegalStateException instead of a NPE, is that expected?